### ISSUE TITLE
Bundle `area`, `sigma_coordinates`, and `timestep` together into a single dataclass

### ISSUE THREAD

**Issue by spencerkclark** [ai2cm/ace#354]
We frequently pass the `area`, `sigma_coordinates`, and `timestep` throughout ACE as a group of individual arguments.  It might be nice to consolidate these into a single dataclass that we could pass as a single argument.  It would also make it easier to potentially attach / pass around similar kinds of information in the future.

xref: https://github.com/ai2cm/full-model/pull/830#discussion_r1608417112

**Comment by mcgibbon** [comment]
Note HorizontalCoordinates is never used and won't be included on this type, at least initially. It's probably still fine to have returned by the data loader, but we don't want to create a bunch of unused dummy data in tests.

**Comment by oliverwm1** [comment]
Is this resolved by ai2cm/full-model#1351?

**Comment by mcgibbon** [comment]
I don't think so... I think this issue is more about passing coordinates in a container throughout the codebase, and not just internally in the data loading code. GriddedOperations is sort of resolving this, but it's not for sigma_coordinates and timestep which are specifically called out in this issue.

I think we could resolve this by adding time-differencing and vertical integration methods to GriddedOperations and using it instead of these coordinate variables.

**Comment by oliverwm1** [comment]
Sounds good. Yes, I agree bundling the timestep, horizontal coordinates and vertical coordinates into a single container would make sense.

I have never quite understood the purpose of `GriddedOperations` though—why not have the relevant methods (e.g. vertical integral or area-weighted horizontal mean) directly on the coordinate classes?

**Comment by mcgibbon** [comment]
> Sounds good. Yes, I agree bundling the timestep, horizontal coordinates and vertical coordinates into a single container would make sense.
> 
> I have never quite understood the purpose of `GriddedOperations` though—why not have the relevant methods (e.g. vertical integral or area-weighted horizontal mean) directly on the coordinate classes?

Mainly because "area weights" are not a coordinate, but are the basis of all of the gridded operations. But this is perhaps an explanation of the organic growth of this code more than a good design reason. If we construct the area weights privately inside the horizontal coordinates class, it could make sense to put horizontal reductions on that class.

It would be nice maybe for GriddedOperations to be composed of these three types, exposing their methods or providing the attributes (`horizontal`, `vertical`, and `time`) that expose these methods. If we don't often pass these together, we can consider not having a top-level container, but I think the motivation of this issue was that we do (e.g. for aggregators).

**Comment by oliverwm1** [comment]
Is this still an open issue? We have the `DatasetInfo` class which combines these things. @mcgibbon
